File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commends):
    answer=[]
    for n in range(len(commends)):
        i=commends[n][0]
        j=commends[n][1]
        k=commends[n][2]
        new_array=array[i-1:j]
        new_array.sort()
        answer.append(new_array[k-1])
    return answer

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> answerList = new ArrayList<>();
        for (int[] c : commands) {
            int i = c[0] - 1;
            int j = c[1];
            int k = c[2] - 1;
            
            int[] sliceArr = Arrays.copyOfRange(array, i, j);
            Arrays.sort(sliceArr);
            answerList.add(sliceArr[k]);
        }
        
        int[] answer = new int[answerList.size()];
        for (int i = 0; i < answerList.size(); i++) {
            answer[i] = answerList.get(i);
        }
        
        return answer;
    }
}